extensions/utils/VaultClientWrapper.js

```diff
@@ -0,0 +1,145 @@
+const http = require('http');
+const https = require('https');
+const { ChainableTemporaryCredentials } = require('aws-sdk');
+const { errorUtils } = require('arsenal');
+
+const { authTypeAssumeRole } = require('../../lib/constants');
+const VaultClientCache = require('../../lib/clients/VaultClientCache');
+const CredentialsManager = require('../../lib/credentials/CredentialsManager');
+
+class VaultClientWrapper {
+    constructor(id, host, port, authConfig, logger) {
+        this._authConfig = authConfig;
+        this._transport = this._authConfig.transport;
+        this._clientId = id;
+        this._vaultHost = host;
+        this._vaultPort = port;
+        this.logger = logger;
+
+        if (this._transport === 'https') {
+            this.stsAgent = new https.Agent({ keepAlive: true });
+        } else {
+            this.stsAgent = new http.Agent({ keepAlive: true });
+        }
+
+        this._tempCredsPromiseResolved = false;
+        this._storeAWSCredentialsPromise();
```

Comment:
I think we should move this method call into the init method as it is part of the init process and is sort of like a pre-init as is